test/index.js

```diff
@@ -94,6 +94,28 @@ describe('KeyringController', function () {
     });
   });
 
+  describe('persistAllKeyrings', function () {
+    it('should persist keyrings in unsupportedKeyrings array', async function () {
+      const unsupportedKeyring = 'DUMMY_KEYRING';
+      keyringController.unsupportedKeyrings = [unsupportedKeyring];
+      await keyringController.persistAllKeyrings();
+      const { vault } = keyringController.store.getState();
+      const keyrings = await mockEncryptor.decrypt(password, vault);
+      expect(keyrings.indexOf(unsupportedKeyring) > -1).toBe(true);
+      expect(keyrings).toHaveLength(2);
+    });
+
+    it('emits "unlock" event', async function () {
+      await keyringController.setLocked();
+
+      const spy = sinon.spy();
+      keyringController.on('unlock', spy);
+
+      await keyringController.submitPassword(password);
+      expect(spy.calledOnce).toBe(true);
+    });
```

Comment:
```suggestion
```

I believe we can remove this since it seems to be a duplicate of the test in `describe('submitPassword'`